modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php

```diff
@@ -68,6 +76,51 @@ public function testAcmsWizardMenu(): void {
       // Ensure that after save button headless mode is enabled.
       $assertSession->pageTextContains('Acquia CMS Pure Headless has been enabled.');
 
+      // Verify administration menus are changes as per headless mode.
+      // Content Menu.
+      $contentMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-system-admin-content');
+      $contentMenuLi = [
+        'Scheduled Content',
+        'Add content',
+        'Files',
+        'Media',
+        'Scheduled Media',
+      ];
+      $this->childMenuTest($contentMenu, 'Content', $contentMenuLi, '.toolbar-icon-system-admin-content');
+      // API Menu.
+      $apiMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-admin-access-control');
+      $apiMenuLi = [
+        'Dashboard',
+        'OAuth clients',
+        'OAuth settings',
+        'Roles',
+        'Tokens',
+        'User accounts',
+      ];
+      $this->childMenuTest($apiMenu, 'API', $apiMenuLi, '.toolbar-icon-admin-access-control');
+      // Data Model Menu.
+      $dataModelMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-admin-content-models');
+      $dataModelMenuLi = [
+        'Content types',
+        'Media types',
+        'Menus',
+        'Taxonomy',
+      ];
+      if ($this->moduleHandler->moduleExists('block_content')) {
+        $dataModelMenuLi[] = 'Custom blocks';
+      }
+      $this->childMenuTest($dataModelMenu, 'Data model', $dataModelMenuLi, '.toolbar-icon-admin-content-models');
+      // System administration Menu.
+      $systemAdministrationMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-admin-cms');
+      $systemAdministrationMenuLi = [
+        'Structure',
+        'Extend',
+        'Configuration',
+        'People',
+        'Reports',
+      ];
+      $this->childMenuTest($systemAdministrationMenu, 'System administration', $systemAdministrationMenuLi, '.toolbar-icon-admin-cms');
```

Comment:
@rajeshreeputra - can we data provider for each menu and pass argument as expected child to reduce the duplicity of code?